source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,467 @@
+``super()`` constructor.
+
+Instead, we are required to provide a ``doesDataExist()`` method. Since the abstraction does not know how to
+obtain the data, it leaves this function to be implemented. As the name says, the method should check if the data
+already exists on the supported target. For the ``HealthDataProcessor``, this always returns true, since every
+living entity always has health.
+
+.. code-block:: java
+
+    protected boolean doesDataExist(EntityLivingBase entity) {
+        return true;
+    }
+
+Setter Methods
+~~~~~~~~~~~~~~
+
+A setter method receives a ``DataHolder`` of some sort and some data that should be applied to it, if possible.
+
+The ``DataProcessor`` interface defines a ``set()`` method accepting a ``DataHolder`` and a ``DataManipulator``
+which returns a ``DataTransactionResult``. Depending on the abstraction class used, some of the necessary
+functionality might already be implemented.
+
+In this case, the ``AbstractEntityDataProcessor`` takes care of most of it and just requires a method to set
+some values to return ``true`` if it was successful and ``false`` if it was not. All checks if the
+``DataHolder`` supports the ``Data`` is taken care of, the abstract class will just pass a Map mapping each
+``Key`` from the ``DataManipulator`` to its value and then construct a ``DataTransactionResult`` depending on
+whether the operation was successful or not.
+
+.. code-block:: java
+
+    protected boolean set(EntityLivingBase entity, Map<Key<?>, Object> keyValues) {
+        entity.getEntityAttribute(SharedMonsterAttributes.maxHealth)
+            .setBaseValue(((Double) keyValues.get(Keys.MAX_HEALTH)).floatValue());
+        entity.setHealth(((Double) keyValues.get(Keys.HEALTH)).floatValue());
+        return true;
+    }
+
+.. tip::
+
+    To understand ``DataTransactionResult`` \ s, check the :doc:`corresponding docs page
+    <../../plugin/data/transactions>` and refer to the `DataTransactionBuilder API
+    Docs <https://jd.spongepowered.org/index.html?org/spongepowered/api/data/DataTransactionBuilder.html>`_ to
+    create one.
+
+Removal Methods
+~~~~~~~~~~~~~~~
+
+The ``remove()`` method attempts to remove data from the ``DataHolder`` and returns a ``DataTransactionResult``.
+
+Since it is impossible for an ``EntityLivingBase`` to not have any health, this operation will always fail on
+the ``HealthDataProcessor``.
+
+.. code-block:: java
+
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        return DataTransactionBuilder.failNoData();
+    }
+
+
+Getter Methods
+~~~~~~~~~~~~~~
+
+Getter methods obtain data from a ``DataHolder`` and return an optional ``DataManipulator``. The
+``DataProcessor`` interface specifies the methods ``from()`` and ``createFrom()``, the difference being that
+``from()`` will return ``Optional.empty()`` if the data holder is compatible, but currently does not contain the
```

Comment:
contain is checked by the implementation of `doesDataExist(T)` should probably mention that as it can be easy to trip up on.